### ISSUE TITLE
Would like to override behavior of MyIdentityManager

### ISSUE THREAD

**Issue by lorenh** [jimsch/CoAP-CSharp#32]
Currently there is an internal class inside of DtlsServer used to look up user keys called MyIdentityManager.  I would like a way to override the behavior to supply a different user key storage mechanism.  Currently this is not possible since it is created directly on line 24 of DtlsServer.cs in the constructor.  An alternate implementation of DtlsServer is also not able to be injected since it is created directly at line 134 of DTLSSession.cs.  Just wanted to mention this in case you are working on some refactoring in this area for the next version.

**Comment by jimsch** [comment]
That is currently no on my to do list and is unlikely to get there until after a get a basic ACL system working.  The key set can be dynamically updated if that is what you are trying to do.  On the other hand a pull request would be interesting as well.

**Comment by lorenh** [comment]
Yes, we did figure out that the KeySet can be updated, and that will get us by for now.  Thanks for letting me know you don't have it on your radar.  I may do a PR when it gets to the point that we need this.

**Comment by jimsch** [comment]
There is now - only in the tls_certs branch in my repro - a set of events that will be triggered if an unknown PSK is found.  Would also fire for certificates and unknown RPKs.  Please look and see if that would solve your needs.
